python/cudf/cudf/core/column/categorical.py

```diff
@@ -318,12 +318,22 @@ def set_base_children(self, value):
     def children(self):
         if self._children is None:
             codes_column = self.base_children[0]
+            if self._mask is None and self.base_mask is not None:
+                mask_column = self.base_children[0].set_mask(self.base_mask)
+
+                self._mask = mask_column[
+                    self.offset : self.offset + self.size
+                ].mask
```

Comment:
Couldn't we just do `self.mask` here and let the implementation in `column.pyx` handle this?

Reply:
Also, should the `children` accessor be masked? It makes sense for `codes` accessor to be masked, but not for `children` I think.

Reply:
Calling directly `self._mask` creates a circular dependency as codes gets called in column.pyx.

Reply:
However because of earlier changes, this change didn't require.